### PR TITLE
NOOS-914/ai-pr-review

### PR DESCRIPTION
**Pull request by zyassine**
**SEMVER Update Type:**
- [x] Major
- [ ] Minor
- [ ] Patch

## Motivation:

 **Closes Issues:**
-  [ISSUE URL](https://noosenergy.atlassian.net/browse/NOOS-914)

## Checklist:

- [x] All new jobs, commands, executors, parameters have descriptions.
- [x] Usage Example version numbers have been updated.
- [x] Changelog has been updated.

### CODE REVIEW

File: .circleci/test-deploy.yml [hunk 1]
```diff
@@ -182,6 +182,15 @@ jobs:
           workspace: noos-prod
           plan_message: ${CIRCLE_BUILD_URL}
 
+  integration_test_trigger_pr_review:
```
Comment (pieeri) [inline .circleci/test-deploy.yml:185]: 🐼 

Maybe reuse the command name in the integration test for clarity, similarly to other tests?

`integration_test_github_post_comment`

File: src/commands/github_post_comment.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+---
+
+description: Post a comment under a github PR.
+
+parameters:
+  comment:
```
Comment (pieeri) [inline src/commands/github_post_comment.yml:6]: 🐼 

What about adding a value by default, e.g. `/review`?

File: .github/workflows/pr_agent_template.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      pull-requests: write
+      contents: write
+    steps:
+      - name: PR Agent action step
+        id: pragent
+        uses: Codium-ai/pr-agent@main
+        env:
+          pr_reviewer.enable_review_labels_security: false
+          pr_reviewer.enable_review_labels_effort: false
+          OPENAI_KEY: ${{ inputs.openai_key }}
```
Comment (pieeri) [inline .github/workflows/pr_agent_template.yml:28]: ❓ 

Given such secrets are shared across the Github organisation for our repositories, is it necessary to template their value?

**Review by pieeri — CHANGES_REQUESTED** [review]
🏓

File: src/commands/github_post_comment.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+          PR_NUMBER=$(echo "${CIRCLE_PULL_REQUEST}" | awk -F '/' '{print $NF}')
+          BASE_URL="https://api.github.com/repos/noosenergy/${CIRCLE_PROJECT_REPONAME}"
+
+          set -- "$@" -X POST
+
+          set -- "$@" -H "Accept: application/vnd.github+json"
+
+          set -- "$@" -H "Authorization: Bearer ${GITHUB_TOKEN}"
+
+          set -- "$@" -H "X-GitHub-Api-Version: 2022-11-28"
```
Comment (pieeri) [inline src/commands/github_post_comment.yml:24]: 🐼 

Maybe as suggested by the AI PR agent, what about being able to pass the Github API version as a parameter?

File: .circleci/test-deploy.yml [hunk 5]
```diff
@@ -16,6 +16,10 @@ release-filters: &release-filters
   branches:
     ignore: /.*/
 
+filters_non_master_only: &filter_non_master_only
```
Comment (pieeri) [inline .circleci/test-deploy.yml:19]: 🐼 

Maybe opportunistic, but good to align CircleCI YAML conventions for `filters_non_master_only` & `release-filters`!

**Review by pieeri — CHANGES_REQUESTED** [review]
Eventually the PR Agent is alive! Thanks for throwing the time in to make it work!

**Review by pieeri — APPROVED** [review]
Let's ship!

**Review by mthiboust — APPROVED** [review]
Let's use it now!

**Pull request merged**